src/composition/watchers/watcher/supergraph_config.rs

```diff
@@ -63,29 +65,86 @@ pub struct SupergraphConfigDiff {
 }
 
 impl SupergraphConfigDiff {
+    /// Compares the differences between two supergraph configs,
+    /// returning the added and removed subgraphs.
     pub fn new(
         old: &SupergraphConfig,
         new: SupergraphConfig,
     ) -> Result<SupergraphConfigDiff, ConfigError> {
         let old_subgraph_defs = old.get_subgraph_definitions().tap_err(|err| {
-            eprintln!(
-                "Error getting subgraph definitions from the current supergraph config: {:?}",
+            // TODO: why do we print here instead of just defering to the caller?
+            errln!(
+                "error getting subgraph definitions from the current supergraph config: {:?}",
                 err
             )
         })?;
```

Comment:
the `?` will defer to the caller; the `tap_err` just adds some visibility for us